Signal/src/ViewControllers/BlockListViewController.m

```diff
@@ -0,0 +1,235 @@
+    OWSAssert(cell);
+    
+    switch (indexPath.section) {
+        case BlockListViewControllerSection_Add:
+            cell.textLabel.text = NSLocalizedString(
+                                                    @"SETTINGS_BLOCK_LIST_ADD_BUTTON", @"A label for the 'add phone number' button in the block list table.");
+            cell.textLabel.font = [UIFont ows_mediumFontWithSize:18.f];
+            cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+            break;
+        case BlockListViewControllerSection_BlockList: {
+            NSString *phoneNumber = _blockedPhoneNumbers[(NSUInteger) indexPath.item];
+            PhoneNumber *parsedPhoneNumber = [PhoneNumber tryParsePhoneNumberFromUserSpecifiedText:phoneNumber];
+            // Try to parse and present the phone number in E164.
+            // It should already be in E164, so this should always work.
+            // If an invalid or unparsable phone number is already in the block list,
+            // present it as-is.
+            cell.textLabel.text = (parsedPhoneNumber
+                                   ? parsedPhoneNumber.toE164
+                                   : phoneNumber);
+            cell.textLabel.font = [UIFont ows_mediumFontWithSize:18.f];
+            cell.accessoryType = UITableViewCellAccessoryCheckmark;
+            break;
+        }
+        default:
+            OWSAssert(0);
+            return 0;
+    }
+    
+    return cell;
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    [tableView deselectRowAtIndexPath:indexPath animated:YES];
+
+    switch (indexPath.section) {
+        case BlockListViewControllerSection_Add:
+        {
+            AddToBlockListViewController *vc = [[AddToBlockListViewController alloc] init];
+            NSAssert(self.navigationController != nil, @"Navigation controller must not be nil");
+            NSAssert(vc != nil, @"Privacy Settings View Controller must not be nil");
+            [self.navigationController pushViewController:vc animated:YES];
+            break;
+        }
+        case BlockListViewControllerSection_BlockList: {
+            NSString *phoneNumber = _blockedPhoneNumbers[(NSUInteger)indexPath.item];
+            [self showUnblockActionSheet:phoneNumber];
+            break;
+        }
+        default:
+            OWSAssert(0);
+    }
+}
+
+- (void)showUnblockActionSheet:(NSString *)phoneNumber
+{
+    OWSAssert(phoneNumber.length > 0);
+
+    PhoneNumber *parsedPhoneNumber = [PhoneNumber tryParsePhoneNumberFromUserSpecifiedText:phoneNumber];
+    NSString *displayPhoneNumber = (parsedPhoneNumber ? parsedPhoneNumber.toE164 : phoneNumber);
+
+    NSString *title = [NSString stringWithFormat:NSLocalizedString(@"BLOCK_LIST_UNBLOCK_TITLE_FORMAT",
+                                                     @"A format for the 'unblock phone number' action sheet title."),
```

Comment:
This logic has been pulled out to a separate class in a subsequent PR.  I'll fix this when we get there: https://github.com/WhisperSystems/Signal-iOS/pull/1927#issuecomment-291271458